cpp/modules/deck.gl/core/src/lib/attribute/attribute-manager.cc

```diff
@@ -33,8 +33,8 @@ auto AttributeManager::getNeedsRedraw(bool clearRedrawFlags) -> bool {
 
 void AttributeManager::setNeedsRedraw() { this->_needsRedraw = true; }
 
-void AttributeManager::add(const std::shared_ptr<lumagl::AttributeDescriptor>& descriptor) {
-  this->_descriptors.push_back(descriptor);
+void AttributeManager::add(const lumagl::garrow::AttributeDescriptor& descriptor) {
+  this->_descriptors.push_back(std::move(descriptor));
```

Comment:
Does `std::move` actually do anything here? Wouldn't descriptor need to be "movable"/"rvalue" for it work?

`const lumagl::garrow::AttributeDescriptor&& descriptor)`

The point is that the compiler needs to now that the object is movable. I.e. the return value of a function is moveable.

Copying this object is OK. It is a rare setup situation. No need to optimize for that.